pkg/services/serviceaccounts/manager/service.go

```diff
@@ -150,7 +158,39 @@ func (sa *ServiceAccountsService) CreateServiceAccount(ctx context.Context, orgI
 	if err := validOrgID(orgID); err != nil {
 		return nil, err
 	}
-	return sa.store.CreateServiceAccount(ctx, orgID, saForm)
+
+	var serviceAccount *serviceaccounts.ServiceAccountDTO
+	err := sa.db.InTransaction(ctx, func(ctx context.Context) error {
+		var err error
+		serviceAccount, err = sa.store.CreateServiceAccount(ctx, orgID, saForm)
+		if err != nil {
+			return err
+		}
+
+		user, err := identity.GetRequester(ctx)
+		if err == nil && user != nil && sa.cfg.RBAC.PermissionsOnCreation("service-account") {
```

Comment:
```suggestion
		if err == nil && sa.cfg.RBAC.PermissionsOnCreation("service-account") {
```

Already checked in identity.GetRequester